### PR TITLE
Adds some miscellaneous, missing method documentation.

### PR DESCRIPTION
**Pull request by jamesonwilliams**
In order to add a Checkstyle for MissingJavadocMethod, we need to first
have Javadoc on our public and protected methods.

This commit begins chiseling away at that problem.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PendingMutation.java [hunk 1]
```diff
@@ -365,34 +364,17 @@ public String toString() {
...
          */
-        UPDATE(MutationType.UPDATE),
+        UPDATE,
 
         /**
          * The removal of a previously-created model.
          */
-        DELETE(MutationType.DELETE);
-
-        private final MutationType apiMutationType;
```
Comment (jamesonwilliams) [inline aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PendingMutation.java:380]: All of this stuff was not used. And, it required more Javadocs. So, I just got rid of it, instead.

Comment (TrekSoft) [reply aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PendingMutation.java:380]: Yay for simplifcation!

File: core/src/test/java/com/amplifyframework/core/model/scalar/AWSTimeTest.java [hunk 2]
```diff
@@ -62,9 +73,14 @@ public void convertsToAndFromDate() {
         Date date = cal.getTime();
         AWSTime awsTime = new AWSTime(date);
         assertEquals(date, awsTime.toDate());
-        assertThrows(IllegalStateException.class, () -> awsTime.getOffsetTotalSeconds());
+        assertThrows(IllegalStateException.class, awsTime::getOffsetTotalSeconds);
```
Comment (TrekSoft) [inline core/src/test/java/com/amplifyframework/core/model/scalar/AWSTimeTest.java:76]: Hmm, interesting syntax. Will remember.

File: rxbindings/src/test/java/com/amplifyframework/rx/RxHubBindingTest.java [hunk 3]
```diff
@@ -29,18 +29,24 @@
...
     @Before
     public void setup() {
         this.rxHub = new RxHubBinding();
     }
 
+    /**
+     * A subscriber should receive a value published on the Hub.
+     */
     @Test
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```
Comment (TrekSoft) [inline rxbindings/src/test/java/com/amplifyframework/rx/RxHubBindingTest.java:49]: Yeah this is better down at the method level

**Pull request merged**